specification/azurelargeinstance/resource-manager/Microsoft.AzureLargeInstance/preview/2023-07-20-preview/azurelargeinstance.json

```diff
@@ -0,0 +1,1106 @@
+            "$ref": "./examples/AzureLargeInstance_ListByResourceGroup.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AzureLargeInstanceListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureLargeInstance/azureLargeInstance/{azureLargeInstanceName}/start": {
+      "post": {
+        "tags": [
+          "Azure Large Instances"
+        ],
+        "summary": "Start an Azure Large Instance.",
+        "description": "The operation to start an Azure Large Instance (only for compute instances)",
+        "operationId": "AzureLargeInstance_Start",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location",
+          "final-state-schema": "../../../../../common-types/resource-management/v5/types.json#/definitions/OperationStatusResult"
+        },
+        "x-ms-examples": {
+          "Start an Azure Large Instance": {
+            "$ref": "./examples/AzureLargeInstance_Start.json"
+          }
+        },
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AzureLargeInstanceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/OperationStatusResult"
```

Comment:
Is this response supposed to be returned when finally instance is started? Generally, the properties related to a resource are returned or just confirmation about that it is started.

Reply:
> "Is this response supposed to be returned when finally instance is started? "

It is a confirmation that the process has started. Typically, our machines take a few minutes to restart, so the entire process is asynchronous. On the initial /restart call, we return a 202 status along with the location URL.

> "Also, when one gets 200 response, they expect to see "final" response now, but here operationStatus might show still InProgress kind of result, is it?"

The GET /location provides a 200 response when it successfully retrieves the current state of the operation. The power management process only reaches its final states when the operation status is in one of the following states: "Succeeded", "Failed", or "Canceled".

Reply:
In my opinion, 200 should only be returned when restart operation has completed. Until then, location header should simply return 202 and a location URI pointing to operation result object with details.

Reply:
My mistake, after validating our intenal behaviour I noticed that we are indeed respondoidng 202 until the operation status is in a final state. 

Example of POST call:
<img width="1627" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/26116542/639c2c7e-006f-4ebe-9097-62d1735e70be">

Example in GET responding 202 when in progress:
<img width="1604" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/26116542/d2388459-9e3b-4dd2-bbc1-1b60c3772ba9">

Example GET responding 200 after Succeeded:
<img width="1633" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/26116542/265f90e2-e0b4-4b65-bb94-31f5d0f203d7">

Reply:
Got it. But it is not good to use a standard property definition like OperationStatusResponse which is meant to be used in other scenarios, not for an API response like this. Just FYI.